### PR TITLE
[Feature] Add mmrazor support

### PR DESCRIPTION
**Pull request by wutongshenqiu**
## Motivation

Add [mmrazor](https://github.com/open-mmlab/mmrazor) support.

## Modification

To support mmrazor, only need to add an attribute `from_mmrazor = True` in config file and modify `init_pytorch_model` method in corresponding task. 

## Use cases 
Here we take mmclassification onnx for example.

A minimum independent config is like

```python
onnx_config = dict(
    type='onnx',
    export_params=True,
    keep_initializers_as_inputs=False,
    opset_version=11,
    save_file='end2end.onnx',
    input_names=['input'],
    output_names=['output'],
    input_shape=None)

backend_config = dict(type='onnxruntime')
codebase_config = dict(type='mmcls', task='Classification', from_mmrazor=True)
```

The only defference is an extra `from_mmrazor = True` attribute

```diff
- codebase_config = dict(type='mmcls', task='Classification')
+ codebase_config = dict(type='mmcls', task='Classification', from_mmrazor=True)
```

Besides, `model_cfg`  cound be downloaded [here](https://github.com/open-mmlab/mmrazor/blob/dev_v0.3.0/configs/nas/spos/spos_subnet_shufflenetv2_8xb128_in1k.py) and `checkpoint` [here](https://openmmlab-share.oss-cn-hangzhou.aliyuncs.com/mmrazor/v0.1/nas/spos/spos_shufflenetv2_subnet_8xb128_in1k/spos_shufflenetv2_subnet_8xb128_in1k_flops_0.33M_acc_73.87_20211222-1f0a0b4d.pth?versionId=CAEQHxiBgIDK5b6I7xciIDM1YjIwZjQxN2UyMDRjYjA5YTM5NTBlMGNhMTdkNjI2), to use mmdeploy, model config file should manully add `mutable_cfg` attribute like follows. 

```python
algorithm = dict(
    ...
    mutable_cfg=${MUTABLE_CFG_PATH}
    ...
)
```
The mutable config file could be downloaded [here](https://openmmlab-share.oss-cn-hangzhou.aliyuncs.com/mmrazor/v0.1/nas/spos/spos_shufflenetv2_subnet_8xb128_in1k/spos_shufflenetv2_subnet_8xb128_in1k_flops_0.33M_acc_73.87_20211222-454627be_mutable_cfg.yaml?versionId=CAEQHxiBgICw5b6I7xciIGY5MjVmNWFhY2U5MjQzN2M4NDViYzI2YWRmYWE1YzQx)

## Checklist

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.

### CODE REVIEW

**Comment by grimoire** [comment]
The result is different from mobilenetv2 in mmcls model zoo.
image: https://github.com/grimoire/mmclassification/blob/master/tests/data/color.jpg
mmcls: tabby, tabby cat 0.30
mmrazor: egyptian cat 0.19
Is it expected?

**Comment by wutongshenqiu** [comment]
> The result is different from mobilenetv2 in mmcls model zoo. image: https://github.com/grimoire/mmclassification/blob/master/tests/data/color.jpg mmcls: tabby, tabby cat 0.30 mmrazor: egyptian cat 0.19 Is it expected?

Yep, we obtain the architecture of model by [SPOS](https://arxiv.org/abs/1904.00420), which might be different from original implementation with mmcls. Besides, the result of the checkpoint could be found [here](https://github.com/open-mmlab/mmrazor/tree/dev_v0.3.0/configs/nas/spos#results-and-models)

**Review by AllentDan — COMMENTED** [review]
Test data could be simplified if possible

**Comment by wutongshenqiu** [comment]
Simplified model configs in test data.

File: mmdeploy/codebase/base/task.py [hunk 1]
```diff
@@ -284,3 +284,21 @@ def get_model_name(self) -> str:
...
+    def from_mmrazor(self) -> bool:
+        """Whether the codebase from mmrazor.
+
+        Returns:
+            bool: From mmrazor or not.
+
+        Raises:
+            TypeError: An error when type of `from_mmrazor` is not boolean.
+        """
+        from_mmrazor = getattr(self.deploy_cfg.codebase_config, 'from_mmrazor',
```
[the remark this answers is not in the document]
Comment (wutongshenqiu) [reply mmdeploy/codebase/base/task.py]: done.

**Review by grimoire — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
LGTM after fixing unit tests

**Comment by wutongshenqiu** [comment]
:raised_hands: Unit tests should be possible to fix after pr [open-mmlab/mmrazor#128](https://github.com/open-mmlab/mmrazor/pull/128) is merged.

**Comment by grimoire** [comment]
Don't worry, approval will be dismissed when any change happened. That is expected.

**Comment by grimoire** [comment]
Edit this `https://github.com/open-mmlab/mmdeploy/blob/73cf3b5feb2653ab12858bc1a311125cbc7bda9b/.github/workflows/build.yml#L81` might be a workaround.
It did surprise me that the environment does not include git...

**Comment by wutongshenqiu** [comment]
Got it, I will try that.

**Comment by grimoire** [comment]
Will it be solved in next mmrazor release(we can change the requirement then)?

**Comment by wutongshenqiu** [comment]
Yes!
After mmrazor 0.3.0 is released, it can be installed via pip instead of git

**Comment by wutongshenqiu** [comment]
Sorry for waiting so long.
MMRazor no longer needs to be installed via git, so the problem is supposed to be solved.

**Review by grimoire — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**